controllers/dashboard_controller.go

```diff
@@ -170,23 +170,26 @@ func (r *GrafanaDashboardReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
 
-	// skip dashboards without an instance selector
-	if dashboard.Spec.InstanceSelector == nil {
-		controllerLog.Info("no instance selector found for dashboard, nothing to do", "name", dashboard.Name, "namespace", dashboard.Namespace)
-		return ctrl.Result{RequeueAfter: RequeueDelay}, nil
-	}
-
 	instances, err := GetMatchingInstances(ctx, r.Client, dashboard.Spec.InstanceSelector)
 	if err != nil {
 		controllerLog.Error(err, "could not find matching instance", "name", dashboard.Name)
+		dashboard.Status.NoMatchingInstances = true
+		err = r.Client.Status().Update(ctx, dashboard)
+		if err != nil {
+			controllerLog.Error(err, "unable to update status", "name", dashboard.Name)
+		}
 		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
 
 	if len(instances.Items) == 0 {
 		controllerLog.Info("no matching instances found for dashboard", "dashboard", dashboard.Name, "namespace", dashboard.Namespace)
-
+		dashboard.Status.NoMatchingInstances = true
+		err = r.Client.Status().Update(ctx, dashboard)
+		if err != nil {
+			controllerLog.Error(err, "unable to update status", "name", dashboard.Name)
+		}
 		// TODO when a label selector has been updated to no longer match any Grafana instances, should we delete the dashboard from those instances?
-		return ctrl.Result{Requeue: false}, nil
+		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
 
```

Comment:
Should we add `dashboard.Status.NoMatchingInstances: false` somewhere around this line?